Libs/MRML/Core/vtkMRMLPlotNode.h

```diff
@@ -0,0 +1,249 @@
+/*==============================================================================
+
+  Copyright (c) Kapteyn Astronomical Institute
+  University of Groningen, Groningen, Netherlands. All Rights Reserved.
+
+  See COPYRIGHT.txt
+  or http://www.slicer.org/copyright/copyright.txt for details.
+
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+
+  This file was originally developed by Davide Punzo, Kapteyn Astronomical Institute,
+  and was supported through the European Research Consil grant nr. 291531.
+
+==============================================================================*/
+
+#ifndef __vtkMRMLPlotNode_h
+#define __vtkMRMLPlotNode_h
+
+#include <string>
+#include <vector>
+
+// MRML Includes
+#include "vtkMRMLNode.h"
+
+class vtkMRMLStorageNode;
+class vtkMRMLTableNode;
+
+// VTK Includes
+class vtkColor4ub;
+class vtkPlot;
+class vtkTable;
+
+/// \brief MRML node to represent a Plot object
+///
+class VTK_MRML_EXPORT vtkMRMLPlotNode : public vtkMRMLNode
+{
+public:
+  /// Data types supported by the Plot. Used in qMRMLPlotModel for visualization.
+
+public:
+  static vtkMRMLPlotNode *New();
+  vtkTypeMacro(vtkMRMLPlotNode,vtkMRMLNode);
+
+  // Description:
+  // Enum of the available plot types
+  enum {
+    LINE,
+    POINTS,
+    BAR,
+  };
+
+  void PrintSelf(ostream& os, vtkIndent indent);
+
+  //----------------------------------------------------------------
+  /// Standard methods for MRML nodes
+  //----------------------------------------------------------------
+
+  virtual vtkMRMLNode* CreateNodeInstance();
+
+  ///
+  /// Set node attributes
+  virtual void ReadXMLAttributes( const char** atts);
+
+  ///
+  /// Write this node's information to a MRML file in XML format.
+  virtual void WriteXML(ostream& of, int indent);
+
+  ///
+  /// Copy the node's attributes to this object
+  virtual void Copy(vtkMRMLNode *node);
+
+  ///
+  /// Copy the node's attributes to this object but changing the name and type
+  virtual void CopyAndSetNameAndType(vtkMRMLNode *node, const char *name, int Type);
```

Comment:
I don't think this combination would be such a common operation that it would warrant a convenience function.